.github/workflows/deploy-staging.yml

```diff
@@ -0,0 +1,96 @@
+name: "Deploy - Staging"
+on: 
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Branch name"
+        type: string
+        required: true
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  release-docker:
+    name: "Release Docker image"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Print inputs
+        run: echo "Building docker image from ${{ github.event.inputs.branch }} branch..."
+
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.branch }}
+      
+      - name: Install DigitalOcean CLI
+        uses: digitalocean/action-doctl@v2
+        with:
+          token: ${{ secrets.DIGITALOCEAN_ACCESS_TOKEN }}
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          version: latest
+      
+      - name: Login to DigitalOcean Container Registry
+        run: doctl registry login --expiry-seconds 600
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Configure Docker image metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=ghcr.io/${{ github.repository}}
+            name=registry.digitalocean.com/cheqd/${{ github.repository }}
+          tags: |
+            type=raw,value=staging-latest
+          labels: |
+            org.opencontainers.image.title="cheqd DID Resolver"
+            org.opencontainers.image.description="DID Resolver for the did:cheqd method"
+            org.opencontainers.image.source="https://github.com/cheqd/did-resolver"
+            org.opencontainers.image.vendor="Cheqd Foundation Limited"
+            org.opencontainers.image.created={{date 'dddd, MMMM Do YYYY, h:mm:ss a'}}
+            org.opencontainers.image.documentation="https://docs.cheqd.io/identity"
+
+      - name: Build and push image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: docker/Dockerfile
+          platforms: linux/amd64
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
Do we really need this? We do a Docker build in the build stage. Release should automatically be pushing to Github container registry. So, at this stage, we shouldn't push to gchr.io and to be honest we shouldn't need to rebuild either since we can just push the file made in build stage to DOCR for staging deploy.

Reply:
@filipdjokic Can you take a look and resolve this?

Reply:
I removed pushing to ghcr, there is no need to do that.

I'm not sure about removing the docker build from this phase only because of the stuff we talked about before (when developers are working on multiple branches at the same time).

This is a manually triggered workflow and it wouldn't be easy to find specific branch artifacts and it would require some kind of naming (also I'm not sure if it's possible to share artifacts between workflows).

With the current setup, developers would just enter a branch name they want to deploy, action will checkout to that branch and push with the staging-latest tag.

It will run slightly longer, but I guess it will give us more flexibility.